clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -541,6 +548,7 @@ else if (e instanceof AuthenticationException) // will be logged later as error
         //this may cause starvation of reads when memory is low. to address this we shuffle the keys if memory is low.
         if (!outOfMemory && memoryPool.availableMemory() < lowMemThreshold) {
             List<SelectionKey> shuffledKeys = new ArrayList<>(selectionKeys);
+            log.warn("memory low: " + memoryPool.availableMemory() + " < " + lowMemThreshold);
```

Comment:
Probably fine, but did you pull this in intentionally?

Reply:
this was part of KAFKA-7304:
https://github.com/apache/kafka/pull/5601/files#diff-d71b50516bd2143d208c14563842390aR601